src/deno_json/mod.rs

```diff
@@ -40,6 +40,12 @@ pub use ts::IgnoredCompilerOptions;
 pub use ts::JsxImportSourceConfig;
 pub use ts::TsConfig;
 
+#[derive(Clone, Debug, Hash, PartialEq)]
+pub struct ConfigWithFiles<T> {
+  pub config: T,
+  pub files: FilePatterns,
+}
```

Comment:
I'm going to change everything to use this. Just started with lint first to try it out.